src/bin/extract_metadata.c

```diff
@@ -69,21 +69,41 @@ static int handle_variable_sav(int index, readstat_variable_t *variable, const c
     const char *label = readstat_variable_get_label(variable);
     int decimals = -1;
 
-    if (readstat_variable_get_type_class(variable) == READSTAT_TYPE_CLASS_STRING) {
+    switch (readstat_variable_get_type_class(variable)) {
+    case READSTAT_TYPE_CLASS_STRING:
         type = "STRING";
-    } else if (readstat_variable_get_type_class(variable) == READSTAT_TYPE_CLASS_NUMERIC) {
-        if (format && (strncmp(format, "DATE", sizeof("DATE")-1) == 0 || 
-                    strncmp(format, "ADATE", sizeof("ADATE")-1) == 0 || 
-                    strncmp(format, "EDATE", sizeof("EDATE")-1) == 0 || 
+        break;
+    case READSTAT_TYPE_CLASS_NUMERIC:
+        // Check for date and duration types
+        // https://libguides.library.kent.edu/SPSS/DatesTime
+        if (format && (strncmp(format, "DATE", sizeof("DATE")-1) == 0 ||
+                    strncmp(format, "ADATE", sizeof("ADATE")-1) == 0 ||
+                    strncmp(format, "EDATE", sizeof("EDATE")-1) == 0 ||
                     strncmp(format, "SDATE", sizeof("SDATE")-1) == 0)) {
+            // TODO: Include all remaining date types
             type = "DATE";
+        } else if (format && (strncmp(format, "MTIME5", sizeof("MTIME5")-1) == 0 ||
+                    strncmp(format, "MTIME8.2", sizeof("MTIME8.2")-1) == 0 ||
+                    strncmp(format, "TIME5", sizeof("TIME5")-1) == 0 ||
+                    strncmp(format, "TIME8", sizeof("TIME8")-1) == 0 ||
+                    strncmp(format, "TIME11.2", sizeof("TIME11.2")-1) == 0 ||
+                    strncmp(format, "DTIME9", sizeof("DTIME9")-1) == 0 ||
+                    strncmp(format, "DTIME12", sizeof("DTIME12")-1) == 0 ||
+                    strncmp(format, "DTIME15.2", sizeof("DTIME15.2")-1) == 0)) {
```

Comment:
You might simplify these comparisons with `sscanf`, which can do basic pattern matching. I typically do something like

```C
long width;
if (sscanf(format, "MTIME%ld", &width) == 1) {
} else if (sscanf(format, "TIME%ld", &width) == 1) {
...
```

It's probably okay to be a little loose with the allowed widths and precisions.